pkg/scaffold/resource/resource_test.go

```diff
@@ -71,24 +71,27 @@ var _ = Describe("Resource", func() {
 			Expect(instance.Validate().Error()).To(ContainSubstring("kind cannot be empty"))
 		})
 
-		It("should fail if the Kind is not pascal cased", func() {
-			// Base case
-			instance := &Resource{Group: "crew", Kind: "FirstMate", Version: "v1"}
+		It("should succeed if the Kind is valid, according to core Kubernetes", func() {
```

Comment:
IHMO the existent KInds check should pass. 
Here for example : https://github.com/kubernetes-sigs/kubebuilder/issues/1019 is for we allow AWSCluster

Reply:
Agreed; this PR relaxes `Kind` validation (to bring it in line with the way it happens in core k8s). Previous passing tests continue to pass, and some previously failing should no longer fail (e.g. [this one](https://github.com/kubernetes-sigs/kubebuilder/blob/971cf03ea49aefc2380fdb6c5a91c55316f4617e/pkg/scaffold/resource/resource_test.go#L83-L91)). Or am I missing something? 🤔

Reply:
The validation obtained is not the correct one for we check the kinds. If you use the correct one the only change here will be adding more one test to ensure that `AWSCluster` will pass. Also, note that is required to remove the pascalze as well since it is not something check-in k8s API too.